src/http2_stream.erl

```diff
@@ -546,3 +564,21 @@ rst_stream_(ErrorCode,
                       RstStream}),
     sock:send(Socket, RstStreamBin),
     ok.
+
+check_content_length(Stream) ->
+    ContentLength =
+        proplists:get_value(<<"content-length">>,
+                            Stream#stream_state.request_headers),
+
+    case ContentLength of
+        undefined ->
+            ok;
+        _Other ->
+            case Stream#stream_state.request_body_size =:= binary_to_integer(ContentLength) of
```

Comment:
if content-length was invalid (a letter or something), would it have been caught before getting here?